src/flock.c

```diff
@@ -171,13 +176,15 @@ int main(int argc, char *argv[]) {
     { "nb",         no_argument,            NULL,           'n' },
     { "wait",       required_argument,      NULL,           'w' },
     { "timeout",    required_argument,      NULL,           'w' },
+    { "conflict-exit-code",    required_argument,      NULL,           'E' },
```

Comment:
does some other implementation have the longopt `--conflict-exit-code`? Seems long winded.

We probably don't need as many spaces between the fields in this row, we can at least line up `NULL,` and `'E' },`

Reply:
The flock implemented by [util-linux supports "--conflict-exit-code"](https://github.com/karelzak/util-linux/blob/f45f3ec34a37e2779d09625b93be5eb91174f8fc/sys-utils/flock.c#L67) as does [FreeBSD's](https://www.freebsd.org/cgi/man.cgi?query=flock). This is option is pretty useful and can be critical in cases where you want to differentiate between an expected failure caused by lock contention versus option misuse or a problem with the flock implementation. Here's a real-world use-case: for a machine at work, I wrote a script that does some real(ish)time post-processing of log files. I use cron with "* * * * *" in lieu of a watchdog or a proper init/service script (sue me `:P`), and if the cron job fails to run the script because of lock contention, that's not really an error for the particular use case, so "--conflict-exit-code" can be used with a "magic" exit status that the parent script can use to decide whether it should fail (flock couldn't be run due to a logic error or system problem) or succeed (there's another process already running which isn't an error in this particular situation). For example:

```
(
    flock --conflict-exit-code=255 -n 9 || exit "$(($? != 255))"
    # ...
) 9>"$lock_file"
```